sql/src/main/java/io/crate/execution/engine/window/WindowBatchIterator.java

```diff
@@ -67,11 +68,11 @@
 
     private final BatchIterator<Row> source;
     private final List<WindowFunction> functions;
-    private final WindowDefinition windowDefinition;
     private final Object[] outgoingCells;
     private final LinkedList<Object[]> standaloneOutgoingCells;
     private final LinkedList<Object[]> resultsForCurrentFrame;
     private final List<CollectExpression<Row, ?>> standaloneExpressions;
+    private final BiFunction<Object[], Object[], Boolean> peerCellsValidator;
```

Comment:
Maybe use a `BiPredicate` - or does the return value have to be nullable?

Maybe also consider to rename it to something like `arePeerCells`.   `Validator` is a bit vague in that it leaves the question what exactly it validates open and what's considered valid or not.

Reply:
Yep, good points